### PR TITLE
feat(init): #MA-997 add initialization for 1d and 2d (massmailing)

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
L'objectif du ticket MA-997 est de séparé le bouton d'initialisation de presence en 2 bouton d'initialisation, 1 pour les droits 1d, l'autre pour les droit 2d. C'est 2 bouton sont lier a un droit et cela va permettre avoir des données d'initialisation différent entre les établissements 1d et 2d

Doc: https://confluence.support-ent.fr/display/MP/Massmailing

## Checklist tests
Initialiser un établisement 1d
Initialiser un établisement 2d

## Issue ticket number and link
[MA-997](https://jira.support-ent.fr/browse/MA-997)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by adrienFro — APPROVED** [review]

File: massmailing/src/main/java/fr/openent/massmailing/helper/init/IInitMassmailingHelper.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import fr.openent.presences.model.Settings;
+
+import java.util.List;
+
+public interface IInitMassmailingHelper {
+    Init1DMassmailingHelper init1DMassmailingHelper = new Init1DMassmailingHelper();
+    Init2DMassmailingHelper init2DMassmailingHelper = new Init2DMassmailingHelper();
+
+    static IInitMassmailingHelper getInstance(InitTypeEnum initTypeEnum) {
+        if (initTypeEnum == InitTypeEnum.ONE_D) {
```
Comment (flmartineau) [inline massmailing/src/main/java/fr/openent/massmailing/helper/init/IInitMassmailingHelper.java]: return (initTypeEnum == InitTypeEnum.ONE_D) ? init1DMassmailingHelper : init2DMassmailingHelper

File: massmailing/src/main/java/fr/openent/massmailing/helper/init/Init1DMassmailingHelper.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.List;
+
+public class Init1DMassmailingHelper implements IInitMassmailingHelper {
+    protected Init1DMassmailingHelper() {
+    }
+
+    @Override
+    public List<Template> getTemplates() {
+        List<Template> templateList = new ArrayList<>();
+        for (int i = 0; i < 2; i++) {
```
Comment (flmartineau) [inline massmailing/src/main/java/fr/openent/massmailing/helper/init/Init1DMassmailingHelper.java]: Met le 2 dans une variable

File: massmailing/src/main/java/fr/openent/massmailing/helper/init/Init2DMassmailingHelper.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.List;
+
+public class Init2DMassmailingHelper implements IInitMassmailingHelper {
+    protected Init2DMassmailingHelper() {
+    }
+
+    @Override
+    public List<Template> getTemplates() {
+        List<Template> templateList = new ArrayList<>();
+        for (int i = 0; i < 2; i++) {
```
Comment (flmartineau) [inline massmailing/src/main/java/fr/openent/massmailing/helper/init/Init2DMassmailingHelper.java]: pareil

File: massmailing/src/main/java/fr/openent/massmailing/service/impl/DefaultInitService.java [hunk 4]
```diff
@@ -10,39 +15,28 @@
...
-                    .add(smsName)
-                    .add(smsContent)
-                    .add("SMS")
-                    .add(owner)
-                    .add(smsCategory);
-        }
-
         query = query.substring(0, query.length() - 1) + ";";
         handler.handle(new Either.Right<>(new JsonObject()
                 .put("statement", query)
```
Comment (flmartineau) [inline massmailing/src/main/java/fr/openent/massmailing/service/impl/DefaultInitService.java]: magic string

**Review by flmartineau — APPROVED** [review]

**Pull request merged**